### PR TITLE
Refactor break blocks to use calculated mining time

### PR DESCRIPTION
**Pull request by nathanruiz**
I've added basic support for breaking blocks. At the moment, it can calculate the time required to mine each block for each tool with these exceptions:
- Using swords: Cobwebs, cocoa, leaves, melons, pumpkins, vines
- Using shears: Cobwebs, wool, leaves, vines

As part of this, I moved some of the existing code from `Server` into an ECS system to make it easier to maintain.

### CODE REVIEW

**Comment by nathanruiz** [comment]
@terrarier2111 I've run the formatter, checks are passing now.

File: src/server/mod.rs [hunk 1]
```diff
@@ -2664,3 +2478,14 @@ fn calculate_relative_teleport(flag: TeleportFlag, flags: u8, base: f64, val: f6
         base + val
     }
 }
+
+pub struct Network {
```
Comment (terrarier2111) [inline src/server/mod.rs:2482]: This doesn't seem very well fledged out, why not put the version inside the Conn directly or pass it when calling connect?

**Review by terrarier2111 — CHANGES_REQUESTED** [review]
Other than the network thingy this change looks great, thank you!

File: src/server/mod.rs (hunk 1 above)
Comment (nathanruiz) [reply src/server/mod.rs:2482]: Ah, I didn't know Conn contained protocol_version. I've switch a few of the generalized protocol operations to use the version from Conn instead.

**Pull request merged**